eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/core/Box.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.core;
+
+import android.os.Parcelable;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public interface Box<T> extends Parcelable
+{
+
+    T content();
```

Comment:
Hmm ... good point. I guess one solution would be to use the DateTimeBox internally instead of the date. Like so:

```
  private Box<DateTime> mDateBox;

  public Dated(DateTime date, EventFilter delegate)
  {
     this(new DateTimeBox(date), delegate);
  }

  private Dated(DateTimeBox datebox, EventFilter delegate)
  {
     mDateBox = datebox;
     mDelegate = delegate;
  }

  public void writeToParcel(Parcel dest, int flags)
    {
        dest.writeParcelable(mDelegate, flags);
        dest.writeParcelable(mDateBox, flags);
    }

        public Dated createFromParcel(Parcel in)
        {
            ClassLoader classLoader = getClass().getClassLoader();
            return new Dated(
                    (DateTimeBox) in.readParcelable(classLoader),
                    (EventFilter) in.readParcelable(classLoader));
        }
```

Also, how would that look if `Dated` doesn't implement `Box` but `Boxable` instead? Shouldn't that fix it too?